### DISCUSSION TITLE
Test questions

### DISCUSSION THREAD

**Discussion by 99alfie** [TrixterFilm/trackteroid#67] [General]
Just wanting to confirm if this is a bug here before opening a ticket:

```
from trackteroid import *
assetversion = Query(AssetVersion).get_first(projections=["components"])
print(assetversion.components)
> EmptyCollection[Component]
print(assetversion.query)
> select id, asset.name, version from AssetVersion where (id is "0010986d-42af-4cde-9ec1-a3b327f80a43")
print(assetversion.query.projections)
> []
```
the query as well as the projections should contain "components"

**Comment by 99alfie** [comment]
these functions do not exist anymore:

Query(AssetVersion).get_one().**get_shot()**
Query(AssetVersion).get_one().**get_task()**

...being used by test_relative_projections in test_query

please advise if there is an alternative or if we should skip the test instead

**Reply by rkoschmitzky** [reply]
If I recall correctly relative terminators should be handled like this now:

```python
Query(AssetVersion).get_one().get(Shot)
Query(AssetVersion).get_one().get(Task)
```

I think this information isn't contained yet in the docs, so it would be worth creating a ticket for updating the documentation for the  _relative terminators_ feature.

**Comment by rkoschmitzky** [comment]
> Just wanting to confirm if this is a bug here before opening a ticket:
> 
> ```
> from trackteroid import *
> assetversion = Query(AssetVersion).get_first(projections=["components"])
> print(assetversion.components)
> > EmptyCollection[Component]
> print(assetversion.query)
> > select id, asset.name, version from AssetVersion where (id is "0010986d-42af-4cde-9ec1-a3b327f80a43")
> print(assetversion.query.projections)
> > []
> ```
> 
> the query as well as the projections should contain "components"

Can you please change the logging level so we can see the INFO output of the performed query?

**Reply by 99alfie** [reply]
Thank you, here is the output (projections do work ok but they do not seem to be stored in `entity.query.projections`)

``` 
INFO:trackteroid.query:Performing query: "select id, version, components, asset.name from AssetVersion limit 1 order by id"
```

**Reply by rkoschmitzky** [reply]
I think it should be considered to not expose `.query` directly anymore and make it a protected member instead as it is only required and primarily useful for internal usage (as we have to keep track of the used session and relationships schema). 
Also, a `.query` instance [can be invalid](https://github.com/TrixterFilm/trackteroid/blob/main/src/trackteroid/entities/base.py#L1029) depending on which collection it is being retrieved.

There might be some implications with keeping the original projections, that I can not fully remember right now. Is that appearing to be a problem somewhere that the projections are missing in the `.query` instance or is this just a matter of not confusing the user?

**Reply by 99alfie** [reply]
there is a test, `test_projections_resolutions`, which compares it (and fails therefore):

```python
        assert [AssetVersion.relationship[ComponentLocation].relation + ".resource_identifier"] + \
               AssetVersion.projections == assetversion_collection.query.projections
```
so I wonder if we can just skip/modify the test or if that is really an issue that needs fixing
